src/toil/leader.py

```diff
@@ -50,15 +52,54 @@ def __init__(self, jobStore):
                               args=(jobStore, self._stop))
         self._worker.start()
 
-    @staticmethod
-    def statsAndLoggingAggregator(jobStore, stop):
+    @classmethod
+    def writeLogFiles(cls, jobNames, jobLogList, config):
+        def createName(logPath, jobName, logExtension):
+            logName = jobName.replace('-', '--')
+            logName = logName.replace('/', '-')
+            logName = logName.replace(' ', '_')
+            logName = logName.replace("'", '')
+            counter = 0
+            while True:
+                suffix = str(counter).zfill(3) + logExtension
+                fullName = os.path.join(logPath, logName + suffix)
+                if not os.path.exists(fullName):
+                    return fullName
+                counter += 1
+
+        mainFileName = jobNames[0]
+        path = None
+        writeFn = None
+        extension = '.log'
+        if config.writeLogs:
+            path = config.writeLogs
+            writeFn = open
+        if config.writeLogsGzip:
```

Comment:
These two options are are mutually exclusive, right? If so, I would make second `if` an `elif` and `assert not (config.writeLogs and config.writeLogsGzip)` above.